SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/AlwaysInterestedTest/AlwaysInterestedTest.cpp

```diff
@@ -0,0 +1,220 @@
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "Net/UnrealNetwork.h"
+
+/**
+ * This test tests that AlwaysInterested UProperties are replicated to clients and servers correctly.
+ *
+ * The test should include two servers and two clients
+ * The flow is as follows:
+ *  - Setup:
+ *    - Server 1 spawns an actor with AlwaysInterested properties, and two test actors, one of which is set to be AlwaysInterested
+ *  - Test:
+ *    - Move the test actors to server 2 authority area, to ensure interest on server 1 is lost via authority.
+ *    - Validate actors are correctly in or out of view for auth and non-auth servers, and owning and non-owning clients.
+ *    - Move the test actors to server 1 authority area, to ensure interest on server 2 is lost via authority.
+ *    - Validate actors are correctly in the view of auth and non-auth servers.
+ *  - Cleanup:
+ *    - Destroy the actors
+ */
+
+AAlwaysInterestedTest::AAlwaysInterestedTest()
+	: Super()
+{
+	Author = "Mike";
+	Description = TEXT("Test Always Interested Functionality");
+}
+
+void AAlwaysInterestedTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, ActorWithAlwaysInterestedProperty);
+	DOREPLIFETIME(ThisClass, InterestedInThisReplicatedActor);
+	DOREPLIFETIME(ThisClass, NotInterestedInThisReplicatedActor);
+}
+
+void AAlwaysInterestedTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	const float StepTimeLimit = 5.0f;
+
+	{ // Step 0 - Cache worker positions
+		AddStep(TEXT("AlwaysInterested_SpawnActors"), FWorkerDefinition::AllServers, nullptr, [this]() {
+			ULayeredLBStrategy* RootStrategy = GetLoadBalancingStrategy();
+			UAbstractLBStrategy* DefaultStrategy = RootStrategy->GetLBStrategyForLayer(SpatialConstants::DefaultLayer);
+			UGridBasedLBStrategy* GridStrategy = Cast<UGridBasedLBStrategy>(DefaultStrategy);
+			AssertIsValid(GridStrategy, TEXT("Invalid LBS"));
+			const UGridBasedLBStrategy::LBStrategyRegions WorkerRegions = GridStrategy->GetLBStrategyRegions();
+			const VirtualWorkerId LocalWorker = GridStrategy->GetLocalVirtualWorkerId();
+
+			auto GetWorkerPosition = [&](bool bLocalWorker) -> FVector {
+				for (const auto& WorkerRegion : WorkerRegions)
+				{
+					if ((bLocalWorker && WorkerRegion.Key == LocalWorker) || (!bLocalWorker && WorkerRegion.Key != LocalWorker))
+					{
+						const FVector2D Centre = WorkerRegion.Value.GetCenter();
+						return FVector{ Centre.X, Centre.Y, 0.f };
+					}
+				}
+				return {};
+			};
+
+			LocalWorkerPosition = GetWorkerPosition(true);
+			OtherWorkerPosition = GetWorkerPosition(false);
+
+			FinishStep();
+		});
+	}
+
+	{ // Step 1 - Spawn actors on server 1
+		AddStep(TEXT("AlwaysInterested_SpawnActors"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			AssertTrue(HasAuthority(), TEXT("Server 1 requires authority over the test actor"));
```

Comment:
Nit : could run that on all servers, and branch on HasAuthority. That's one less dependency on position.